client/src/components/Map.js

```diff
@@ -33,25 +33,37 @@ class Map extends Component {
     const canvas = this.refs.canvas
     const ctx = canvas.getContext('2d')
 
-    let gameStateReq = await window.fetch('http://localhost:3006/zones')
+    let gameStateReq = await window.fetch('http://localhost:3006/gameStates/0')
+    let zoneReq = await window.fetch('http://localhost:3006/zones/0')
+    let factionReq = await window.fetch('http://localhost:3006/factions/0')
+
     let gameState = await gameStateReq.json()
+    let zones = await zoneReq.json()
+    let factions = await factionReq.json()
+
     this.gameState = gameState
+    this.zones = zones
+    this.factions = factions
 
     this.drawMap(ctx)
   }
 
+  checkMapBounds (x, y) {
+    if (x < 0 || x + 1 > this.boardWidth) return false
+    if (y < 0 || y + 1 > this.boardHeight) return false
+    return true
+  }
+
   drawMap (ctx) {
     let color = null
-    let faction = null
-    let zone = null
+    let owner = null
```

Comment:
Looks like these variables are only used in the scope of the inner for-loop and could be constants defined in that loop instead